tests/components/yandex_transport/test_yandex_transport_sensor.py

```diff
@@ -0,0 +1,138 @@
+"""Tests for the yandex transport platform."""
+
+from json import loads
+from pytest import fixture
+
+import homeassistant.components.sensor as sensor
+import homeassistant.util.dt as dt_util
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+from tests.common import (
+    assert_setup_component,
+    async_setup_component,
+    MockDependency,
+    load_fixture,
+)
+
+REPLY = loads(load_fixture("yandex_transport_reply.json"))
+
+
+def get_fake_reply(stop_id):
+    """Return fake reply."""
+    return REPLY
+
+
+@fixture
+def mock_requester():
+    """Create a mock ya_ma module and YandexMapsRequester."""
+    with MockDependency("ya_ma") as ya_ma:
+
+        class MockRequester:
+            """Fake YandexRequester object."""
+
+            def __init__(self, user_agent=None):
+                """Create mock module object, for avoid importing ya_ma library."""
+                self.get_stop_info = get_fake_reply
+
+            def set_new_session(self):
+                """Fake method for reset http session."""
+                pass
+
+        ya_ma.YandexMapsRequester = MockRequester
```

Comment:
I think we want to set the return value of the requester class to the mock requester instance that we then yield. We want to have access in the test to the same requester instance as the code under test gets access to.

Reply:
Although, we don't seem to use the requester instance specifically in the test, so in this case this doesn't really matter. We could just `yield` below.